src/main/kotlin/uk/gov/justice/digital/hmpps/sentenceplan/services/GoalService.kt

```diff
@@ -30,19 +32,12 @@ class GoalService(
     val planEntity = planRepository.findByUuid(planUuid)
       ?: throw Exception("A Plan with this UUID was not found: $planUuid")
 
+    require(goal.areaOfNeed != null && goal.title != null)
```

Comment:
This could be lifted onto the `Goal` class, and have Spring `@Valid` it on the way into the Controller?